tests/integration/assessment/test_ext_hms.py

```diff
@@ -1,17 +1,24 @@
 import dataclasses
 
-from databricks.labs.lsql.backends import CommandExecutionBackend
+import pytest
+
+from databricks.labs.lsql.backends import CommandExecutionBackend, SqlBackend
 from databricks.sdk.service.iam import PermissionLevel
 
 
+@pytest.fixture
+def sql_backend(ws, env_or_skip) -> SqlBackend:
```

Comment:
Why introduce this fixture?

Reply:
This switches the SQL backend to one that is attached to the external HMS. That's what does this bit from the PR description:
> Update the test fixture so that the resources it creates to be crawled are created in the external HMS, not the main one. Without this the assessment crawlers partially fail because the resources they are configured to crawl do not exist on the external HMS.

Reply:
Ah this works indirectly as the fixture is being picked up by the `installation_ctx` fixture? Could you make it more explicit:

1. Rename the fixture to `external_sql_backend`.
2. Expect it in `test_running_real_assessment_job_ext_hms`
3. Use it in the `installation_ctx.replace` to replace the `sql_backed` attribute

Reply:
That won't work: it's too late because by the time the SQL backend is replaced the resources have already been created in the wrong place.

I'll add a comment here explaining that it's important, as well as in the other location where this is being done.

Reply:
> That won't work: it's too late because by the time the SQL backend is replaced the resources have already been created in the wrong place.

To elaborate on this a little bit: I did try something similar to this approach but haven't been able to make all the pieces work, or figure out exactly where it's going wrong. For some reason at least one schema is created on the normal HMS (and not in the external one). The version I tried was:
```python
    main_cluster_id = env_or_skip("TEST_EXT_HMS_NOUC_CLUSTER_ID")
    cluster_id = env_or_skip("TEST_EXT_HMS_CLUSTER_ID")
    ext_hms_ctx = installation_ctx.replace(
        sql_backend = CommandExecutionBackend(ws, cluster_id),
        config_transform = lambda wc: dataclasses.replace(
            wc,
            override_clusters={
                "main": main_cluster_id,
            },
            include_object_permissions=[f"cluster-policies:{cluster_policy.policy_id}"],
        ),
        extend_prompts={
            r"Instance pool id to be set.*": env_or_skip("TEST_INSTANCE_POOL_ID"),
            r".*Do you want to update the existing installation?.*": 'yes',
            r".*connect to the external metastore?.*": "yes",
            r"Choose a cluster policy": "0",
        },
    )
    del installation_ctx  # Use ext_hms_ctx from now on instead of installation_ctx to ensure the external HMS is used.
```
When the resources are in the wrong location, two things happen:
 - Crawlers can fail to locate any resources, because the configuration specifies a database should be searched but it doesn't exist.
 - Subsequent tasks can fail because loading the inventory mistakenly triggers a crawl (because the inventory is empty). This happens with the grants-crawler in particular, leading to workflow failure because the grants-crawler runs on a `tacl` cluster where the isolation mode doesn't allow the fast-crawler to execute.

Reply:
Thanks for the elaborate research! With the comments it is more clear what happens